### ISSUE TITLE
Sqlite: Using column from table being updated in left join predicate generates invalid SQL

### ISSUE THREAD

**Issue by smitpatel** [dotnet/efcore#28886]
Test: Update_with_cross_join_left_join_set_constant

```C#
        AssertExecuteUpdateSql(
            @"UPDATE ""Customers"" AS ""c""
SET ""ContactName"" = 'Updated'
FROM (
    SELECT ""c0"".""CustomerID"", ""c0"".""Address"", ""c0"".""City"", ""c0"".""CompanyName"", ""c0"".""ContactName"", ""c0"".""ContactTitle"", ""c0"".""Country"", ""c0"".""Fax"", ""c0"".""Phone"", ""c0"".""PostalCode"", ""c0"".""Region""
    FROM ""Customers"" AS ""c0""
    WHERE ""c0"".""City"" IS NOT NULL AND (""c0"".""City"" LIKE 'S%')
) AS ""t""
LEFT JOIN (
    SELECT ""o"".""OrderID"", ""o"".""CustomerID"", ""o"".""EmployeeID"", ""o"".""OrderDate""
    FROM ""Orders"" AS ""o""
    WHERE ""o"".""OrderID"" < 10300
) AS ""t0"" ON ""c"".""CustomerID"" = ""t0"".""CustomerID""
WHERE ""c"".""CustomerID"" LIKE 'F%'");
```
Now throws invalid SQL error earlier used to work. Change in PR #28884 
Filed https://www.sqlite.org/forum/forumpost/09862e12c7

**Comment by roji** [comment]
:angry: 

The person on that issue is asking for our schema. We're just like users when we file issues on other repos :rofl:

**Comment by smitpatel** [comment]
> The person on that issue is asking for our schema. We're just like users when we file issues on other repos 🤣

Not really. I don't think there is anything specific to schema there. It is like if some user file issue on us that `string.IndexOf` is not working on SQLServer, we don't ask for schema if they post the query and generated SQL or exception message.

**Comment by roji** [comment]
This seems like it's the same as https://github.com/npgsql/efcore.pg/issues/2478, which I worked around in https://github.com/npgsql/efcore.pg/issues/2478. Note that this limitation blocks some fairly basic scenarios, like [two inner joins](https://github.com/dotnet/efcore/pull/29522/files#diff-dd49dd3114a50ebe8464649c9412791e88070ba735c91d7ec73509efafe51254R1080).

I've posted another comment on [the SQLite issue](https://www.sqlite.org/forum/forumpost/114b18b3ae), to clarify whether they intend to fix this or not. We can simply port my PostgreSQL workaround to SQLite to fix this.

**Comment by roji** [comment]
This doesn't seem to be a problem with MySQL/MariaDB ([the test isn't skipped](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/master/test/EFCore.MySql.FunctionalTests/BulkUpdates/NorthwindBulkUpdatesMySqlTest.cs#L1294)), @lauxjpn can you confirm?

So this seems to be a problem with PostgreSQL and SQLite, and I worked around it in PostgreSQL (https://github.com/npgsql/efcore.pg/pull/2554). We can wait to see if the SQLite folks react on [their issue](https://www.sqlite.org/forum/forumpost/09862e12c7), and if users complain, we can consider bringing the same workaround from PG to SQLite.

**Comment by lauxjpn** [comment]
> This doesn't seem to be a problem with MySQL/MariaDB ([the test isn't skipped](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/master/test/EFCore.MySql.FunctionalTests/BulkUpdates/NorthwindBulkUpdatesMySqlTest.cs#L1294)), @lauxjpn can you confirm?

Yes, there are no issues with the `Update_with_cross_join_left_join_set_constant` test and any version of MySQL and MariaDB that we support.

**Comment by roji** [comment]
Thanks @lauxjpn
